### ISSUE TITLE
Claude API Key not working

### ISSUE THREAD

**Issue by gurugecl** [agentsea/surfkit#100]
Hi, I'm following the advanced tutorial and would like to use Claude for my task. However it will not take my Anthropic API key when I run the following command even though the OpenAI keys work:

surfkit create agent -n robby001 -r process                 

It says the API key is not valid even though I have verified it works fine outside of this app. I passing it in the same manner as the OpenAI keys without keys.

I would also like to use anthropic/claude-3-5-sonnet-20240620 instead of Opus which I assume I can change in the cli/agent.py file or agent.yaml file. Any assistance would be greatly appreciated. Thank you!

**Comment by pbarker** [comment]
Hey @gurugecl, thanks for reporting this. Looks like a regression, it should be fixed in surfkit `0.1.229`.

As far as how to use `anthropic/claude-3-5-sonnet-20240620` it depends on which agent you are using. For instance, if you are using [SurfSlicer](https://github.com/agentsea/surfslicer/tree/main) the supported providers are defined in https://github.com/agentsea/surfslicer/blob/main/agent.yaml#L21. 

When the agent boots it will look at what API keys are available to it and use them according to the preference order of the supported providers. 

If you are wanting to build or change an agent, the available providers are https://github.com/agentsea/mllm/blob/main/mllm/router.py#L40. I need to add some better docs on this aspect.

Let me know if you have any more trouble or questions. Also feel free to jump in the discord if its easier to connect that way. Cheers

**Comment by gurugecl** [comment]
Hi @pbarker, thanks a lot for resolving that and for the additional guidance. So I installed the latest version and am again going through the advanced tutorial. I decided to go with the default "anthropic/claude-3-opus-20240229" model to start with by just providing my Anthropic API key. But when I run the following command using the default prompt:

surfkit solve "Find a cool cat image in the internet" \
    --agent robby001 --device eve001 --tracker hound001 \
    --starting-url "google.com"

I get the following error:

```
taking action...
mouse coordinates: (282, 1006)
ERROR:mllm.router:Retrying mllm.router.Router.chat.<locals>.call_llm in 0.0 seconds as it raised BadRequestError: AnthropicException - {"type":"error","error":{"type":"invalid_request_error","message":"messages.1.content.0.text.text: Input should be a valid string"}} LiteLLM Retried: 2 times, LiteLLM Max Retries: 3.
ERROR:mllm.router:Retrying mllm.router.Router.chat.<locals>.call_llm in 0.0 seconds as it raised BadRequestError: AnthropicException - {"type":"error","error":{"type":"invalid_request_error","message":"messages.1.content.0.text.text: Input should be a valid string"}} LiteLLM Retried: 2 times, LiteLLM Max Retries: 3.
Traceback (most recent call last):
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/main.py", line 1352, in completion
response = anthropic_chat_completions.completion(
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/llms/anthropic.py", line 667, in completion
raise AnthropicError(
litellm.llms.anthropic.AnthropicError: {"type":"error","error":{"type":"invalid_request_error","message":"messages.1.content.0.text.text: Input should be a valid string"}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/tenacity/__init__.py", line 478, in __call__
result = fn(*args, **kwargs)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/mllm/router.py", line 168, in call_llm
response = self.router.completion(model, thread.to_openai())
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 472, in completion
raise e
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 469, in completion
response = self.function_with_fallbacks(**kwargs)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 2351, in function_with_fallbacks
raise e
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 2276, in function_with_fallbacks
raise e
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 2265, in function_with_fallbacks
response = self.function_with_retries(*args, **kwargs)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 2465, in function_with_retries
raise original_exception
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 2406, in function_with_retries
response = original_function(*args, **kwargs)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 537, in _completion
raise e
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/router.py", line 519, in _completion
response = litellm.completion(
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/utils.py", line 3472, in wrapper
raise e
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/utils.py", line 3363, in wrapper
result = original_function(*args, **kwargs)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/main.py", line 2480, in completion
raise exception_type(
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/utils.py", line 9927, in exception_type
raise e
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/litellm/utils.py", line 8616, in exception_type
raise BadRequestError(
litellm.exceptions.BadRequestError: AnthropicException - {"type":"error","error":{"type":"invalid_request_error","message":"messages.1.content.0.text.text: Input should be a valid string"}} LiteLLM Retried: 2 times, LiteLLM Max Retries: 3

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "/Users/gurugecl/Documents/Faunabio/MachineLearning/LLM/FaunaBrain/agentsea/surfhamster/claudeagent/agent.py", line 200, in take_action
response = router.chat(
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/mllm/router.py", line 213, in chat
return call_llm(thread, model, namespace, expect)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/tenacity/__init__.py", line 336, in wrapped_f
return copy(f, *args, **kw)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/tenacity/__init__.py", line 475, in __call__
do = self.iter(retry_state=retry_state)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/tenacity/__init__.py", line 376, in iter
result = action(retry_state)
File "/opt/homebrew/Caskroom/miniforge/base/envs/agentsea/lib/python3.10/site-packages/tenacity/__init__.py", line 419, in exc_check
raise retry_exc from fut.exception()
tenacity.RetryError: RetryError[<Future at 0x111e03dc0 state=finished raised BadRequestError>]
```
